tests/installation/module_registration/register_extensions_and_modules.pm

```diff
@@ -11,12 +11,24 @@
 use base 'y2_installbase';
 use strict;
 use warnings;
-use testapi qw(save_screenshot get_required_var);
+use testapi qw(save_screenshot get_required_var get_var);
 
 sub run {
     my @scc_addons = grep($_, split(/,/, get_required_var('SCC_ADDONS')));
     $testapi::distri->get_module_registration()->register_extension_and_modules([@scc_addons]);
     save_screenshot;
+    $testapi::distri->wait_for_separate_regcode({
```

Comment:
you need to check if the addon is in that list (with a `grep` for example, before running this expensive action, that we want to run just once, then use that result in line 25 below.

Reply:
After we get the registration page, we can input the register code. 
If don't have the page, why do we need check if the addons in the list.
The logic is, first we get the registration page, when check which addons need input the register code.
The page should be checked first.

Reply:
After we get the registration page, we can input the register code. 
If don't have the page, why do we need check if the addons in the list.
The logic is, first we get the registration page, when check which addons need input the register code.
The page should be checked first.

Reply:
this test module is supposed to handle when there are special addons that have additional regcode and also when it is not the case, for the name that is what is does, looks generic,
But if we only use it when there are separate regcode the name needs to be changed and there is not need for a condition.

In other words, we shouldn't wait for that screen if we don't have those special addons, that is why I suggested to move code.

Reply:
Not solved, the test module names doesn't distinguish between the two cases, but we can do that later.